pkg/controller/application/application_controller.go

```diff
@@ -127,7 +128,14 @@ func (r *ReconcileApplication) Reconcile(request reconcile.Request) (reconcile.R
 		},
 		Spec: appsv1.DeploymentSpec{
 			Selector: &metav1.LabelSelector{
-				MatchLabels: map[string]string{"deployment": instance.Name + "-deployment"},
+				MatchLabels: map[string]string{
+					"deployment": instance.Name + "-deployment",
+					ManagedByLabelKey: ManagedBy,
+					ComponentLabelKey: "Backend",
+					PartOfLabelKey: "Nginx application",
+					NameLabelKey: "nginx-01",
```

Comment:
I would Add 
```
InstanceLabelKey: "01"
```
The `name` could be only "nginx", without 01 (not required)

If a cli/UI has some special usage of those labels it could  look like:
`$AppName ($instance) part of $part-of` 
e.g
`Mysql (production) part of Wordpress`

Reply:
Make sense.